seeds/001_users.ts

```diff
@@ -0,0 +1,98 @@
+import * as bcrypt from 'bcrypt';
+import { Knex } from 'knex';
+
+export async function seed(knex: Knex): Promise<void> {
+  // First, delete existing entries to start fresh
+  await knex('user_info').del();
+  await knex('user_passwords').del();
+  await knex('principals').del();
+
+  // Insert seed users
+  const users = await knex('principals').insert([
+    {
+      identity: 'admin@example.com',
```

Comment:
This field should no longer be used. It's from before there was a `principal_identity` table which is now used instead.

Reply:
LOL shite. how does one stay up to date with most current db/schema state? do they just look at what the table headings are when they load the datagrip/tableplus?

Reply:
well, the 'internal api' is really the specification. So it's always more reliable to call the principalService and 'save()' functions instead of directly inserting into the database.

This has the advantage that you get full typing, and it also continuous to work with code changes.

Reply:
Also for additional context, I didn't drop that column right away because *just* in case someone runs into a bug with that migration, it's nice if the old column is still around. So it's a bit of a precaution to keep the old schema around and then later on remove it once it had some time to settle.

Reply:
Makes sense from a backward compatibility perspective! Lol when I worked corporate we announced deprecations like 3 months in advance